runtime/src/in_mem_accounts_index.rs

```diff
@@ -1159,7 +1162,16 @@ impl<T: IndexValue> InMemAccountsIndex<T> {
                                     let slot_list = slot_list
                                         .take()
                                         .unwrap_or_else(|| v.slot_list.read().unwrap());
-                                    disk.try_write(&k, (&slot_list, v.ref_count()))
+                                    disk.try_write(
+                                        &k,
+                                        (
+                                            &slot_list
+                                                .iter()
```

Comment:
at the moment we have to do into on each info. I need to change the try_write to take an iter farther down. I already did it on some deeper functions. I'll come back to this in a follow on pr.

Reply:
I can't push the change through that I wanted. The problem is the inner type on the disk index is:
(Slot, U). So, the iter expected Item=&(Slot, U).
What we have in the calling code here is:
(&Slot, &T).
So, I need a ref to pass to item for each iter that is &(Slot, U), which I don't have. So I have to build a vec to be able to iterate over its refs...
Later changes which will change this api to only pass a single &U instead of slice will do what you want.